Fix: Check child count before applying interrupt

Werninator opened the pull request:
## Description

Currently, an out of bounds error will happen when:
- there's no child node inside the BeehaveTree
- BeehaveTree.enabled = false

Commit:
Fix: Check child count before applying interrupt

So it doesn't throw an error, when:
- there's no child node inside the BeehaveTree
- the BeehaveTree is disabled

bitbrain commented:
Could you also add a test here? https://github.com/bitbrain/beehave/blob/godot-4.x/test/beehave_tree_test.gd

Commit:
Add duck typing check for interrupt method in child

bitbrain committed:
Merge branch 'godot-4.x' into fix_interrupt_error

Commit:
🧪 Add Trees to UnitTestScene to cause errors if fix is not applied

* EmptyTree
  A tree, that is activated and has no children
* OnlyOneActionTree
  A tree with only one action in it
* DeactivatedTree
  A tree that is deactivated
